EU4toV2/Source/Mappers/Ideas/IdeaEffectMapper.cpp

```diff
@@ -32,280 +32,297 @@ mappers::IdeaEffectMapper::IdeaEffectMapper(std::istream& theStream)
 	registerKeyword(std::regex("[a-zA-Z0-9_]+"), [this](const std::string& idea, std::istream& theStream) {
 		IdeaEffects ideaEffects(theStream);
 
-		armyInvestmentIdeas[idea] = ideaEffects.getArmyInvestmentValue();
-		navyInvestmentIdeas[idea] = ideaEffects.getNavyInvestmentValue();
-		commerceInvestmentIdeas[idea] = ideaEffects.getCommerceInvestmentValue();
-		cultureInvestmentIdeas[idea] = ideaEffects.getCultureInvestmentValue();
-		industryInvestmentIdeas[idea] = ideaEffects.getIndustryInvestmentValue();
-
-		armyTechIdeas[idea] = ideaEffects.getArmyTechScoreValue();
-		navyTechIdeas[idea] = ideaEffects.getNavyTechScoreValue();
-		commerceTechIdeas[idea] = ideaEffects.getCommerceTechScoreValue();
-		cultureTechIdeas[idea] = ideaEffects.getCultureTechScoreValue();
-		industryTechIdeas[idea] = ideaEffects.getIndustryTechScoreValue();
-
-		UHLiberalIdeas[idea] = ideaEffects.getUpperHouseLiberalValue();
-		UHReactionaryIdeas[idea] = ideaEffects.getUpperHouseReactionaryValue();
-
-		orderIdeas[idea] = ideaEffects.getOrderInfluenceValue();
-		libertyIdeas[idea] = ideaEffects.getLibertyInfluenceValue();
-		equalityIdeas[idea] = ideaEffects.getEqualityInfluenceValue();
-
-		literacyIdeas[idea] = ideaEffects.getLiteracyLevels();
+		armyIdeas[idea] = ideaEffects.getArmy();
+		navyIdeas[idea] = ideaEffects.getNavy();
+		commerceIdeas[idea] = ideaEffects.getCommerce();
+		cultureIdeas[idea] = ideaEffects.getCulture();
+		industryIdeas[idea] = ideaEffects.getIndustry();
+
+		liberalIdeas[idea] = ideaEffects.getLiberal();
+		reactionaryIdeas[idea] = ideaEffects.getReactionary();
+
+		orderIdeas[idea] = ideaEffects.getOrder();
+		libertyIdeas[idea] = ideaEffects.getLiberty();
+		equalityIdeas[idea] = ideaEffects.getEquality();
+		literacyIdeas[idea] = ideaEffects.getLiteracy();
+
+		slaveryIdeas[idea] = ideaEffects.getSlavery();
+		upper_house_compositionIdeas[idea] = ideaEffects.getUpper_house_composition();
+		vote_franchiseIdeas[idea] = ideaEffects.getVote_franchise();
+		voting_systemIdeas[idea] = ideaEffects.getVoting_system();
+		public_meetingsIdeas[idea] = ideaEffects.getPublic_meetings();
+		press_rightsIdeas[idea] = ideaEffects.getPress_rights();
+		trade_unionsIdeas[idea] = ideaEffects.getTrade_unions();
+		political_partiesIdeas[idea] = ideaEffects.getPolitical_parties();
+
+
 	});
 
 	LOG(LogLevel::Info) << "getting idea effects";
 	parseStream(theStream);
 }
 
 
-int mappers::IdeaEffectMapper::getArmyInvestmentFromIdea(const std::string& ideaName, int ideaLevel) const
+int mappers::IdeaEffectMapper::getArmyFromIdea(const std::string& ideaName, int ideaLevel) const
 {
-	auto idea = armyInvestmentIdeas.find(ideaName);
-	if (idea != armyInvestmentIdeas.end())
+	auto idea = armyIdeas.find(ideaName);
+	if ((idea != armyIdeas.end()) && (ideaLevel >= 7))
```

Comment:
Please add a test making clear that uncompleted ideas provide no change? Same for other items.

Reply:
done. Test with idealevel=7 and idealevel=4 are present for all effects.